components/molecules/NewsletterForm/newsletter-form.tsx

```diff
@@ -1,33 +1,52 @@
 import React, { useEffect, useState } from "react";
 import Image from "next/image";
 import { AiFillCloseCircle } from "react-icons/ai";
+import clsx from "clsx";
 import Button from "components/atoms/Button/button";
 import TextInput from "components/atoms/TextInput/text-input";
 import SaucedLogo from "img/fallbackImageColor.svg";
 import { validateEmail } from "lib/utils/validate-email";
 
+function encode(data: { [key: string]: string }) {
+  return Object.keys(data)
+    .map((key) => encodeURIComponent(key) + "=" + encodeURIComponent(data[key]))
+    .join("&");
```

Comment:
Isn't that a better approach?
```suggestion
const params = new URLSearchParams(Object.entries(data));
return params.toString();
```

Reply:
Yes it is, But I'll take off the function as it is not being used anyways